### PR TITLE
Windows support

### PR DESCRIPTION
**Pull request by desertkun**
- Need [Visual Studio 2015](https://medium.com/dailyjs/how-to-build-v8-on-windows-and-not-go-mad-6347c69aacd4) to build, also a ton of SDKs are required, not sure what exactly
- x64 only (for now)
- `ninja` (required to build) is downloaded automatically
- no `greenstack` support on Windows
- timeouts are made with `timeSetEvent` / `timeKillEvent`
- also some small minor optimizations that do not deserve separate PR

### CODE REVIEW

**Comment by desertkun** [comment]
The last [build](https://travis-ci.org/tbodt/v8py/builds/423742803) went okay, I don’t know why its showing up as failed. Do you know how to retrigger a build?

**Comment by tbodt** [comment]
It's complaining because the code coverage went down, which is because of the extra if statements added to choose between VLAs and heap allocation on not-windows. It's probably OK to merge, though you could add test cases for longer strings and larger number of arguments.

File: v8py/util.h [hunk 1]
```diff
@@ -0,0 +1,8 @@
+
+#ifdef _WIN32
+// v8.h(1680): error C2059: syntax error: '('
+#undef COMPILER
+#define NO_RETURN __declspec(noreturn)
+#else
+#define NO_RETURN __attribute__((noreturn))
+#endif
```
Comment (tbodt) [inline v8py/util.h]: Could this go in v8py.h instead of its own file?

File: v8py/v8py.cpp [hunk 2]
```diff
@@ -102,6 +117,7 @@ static PyMethodDef v8_methods[] = {
 };
 
 PyObject *null_object = NULL;
+uint16_t *string_buffer = NULL;
```
Comment (tbodt) [inline v8py/v8py.cpp]: I think I'd prefer this to be statically allocated, like `uint16_t string_buffer[STRING_BUFFER_SIZE]`

File: v8py/jsfunction.cpp [hunk 3]
```diff
@@ -35,12 +36,24 @@ PyObject *js_function_call(js_function *self, PyObject *args, PyObject *kwargs)
...
-    jss_from_pys(args, argv, context);
+    MaybeLocal<Value> result;
 
     if (!context_setup_timeout(context)) return NULL;
-    MaybeLocal<Value> result = object->CallAsFunction(context, js_this, argc, argv);
-    delete[] argv;
+#ifndef _WIN32
+    // error C2131: expression did not evaluate to a constant on Windows
+    if (argc <= 16) {
+        std::vector< Local<Value> > argv(argc);
```
Comment (tbodt) [inline v8py/jsfunction.cpp]: `std::vector` allocates on the heap, doesn't it? If so there's no performance benefit over a straight heap-allocated array, and therefore no need for this branch and #ifdef.

File: v8py/convert.cpp [hunk 4]
```diff
@@ -38,7 +39,8 @@ PyObject *py_from_js(Local<Value> value, Local<Context> context) {
     if (value->IsArray()) {
         Local<Array> array = value.As<Array>();
         PyObject *list = PyList_New(array->Length());
-        for (unsigned i = 0; i < array->Length(); i++) {
+        uint32_t length = array->Length();
+        for (uint32_t i = 0; i < length; i++) {
```
Comment (tbodt) [inline v8py/convert.cpp]: This seems a little pointless, unless it fixes some compiler error? Also see line 63 and 262 of this file

**Pull request merged**